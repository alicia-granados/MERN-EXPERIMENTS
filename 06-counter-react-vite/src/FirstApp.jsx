const newMessage= 'Fernando';
const newMessage2 ={
    message : 'hola mundo',
    title : 'Fernando'
}
const saludo = () =>{
    return 'Hola como estas'
}
const FirstApp = () => {
    return (
        <>
        <h2> { 1 + 1 } </h2>
        <p>{newMessage} </p>
        <code>{ JSON.stringify(newMessage2)} </code>
        <p>Soy un subtitulo</p>
        <h1>{saludo()}</h1>
        </>
    )
}

export default FirstApp
