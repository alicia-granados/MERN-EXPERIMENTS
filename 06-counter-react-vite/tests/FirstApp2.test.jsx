/** @jest-environment jsdom */
import { render, screen } from "@testing-library/react"
import FirstApp from "../src/FirstApp"

describe('Pruebas en <FirstApp/>', () => {
  
  const tittle='Hola soy goku';
  const subtitle='Hola soy subtitulo'

  test('debe de hacer match con el shapshot',() =>{
    const {container } = render(<FirstApp title={tittle}/>)
    expect ( container ).toMatchSnapshot();
  });

  test('dede de mostrar el mensaje "Hola soy goku"',() =>{
    render(<FirstApp title={tittle}/>)
    expect(screen.getByText(tittle)).toBeTruthy();
    //screen.debug();
  });

  test('dede de mostrar el titulo en un h1',() =>{
    render(<FirstApp title={tittle}/>)
    //expect( screen.getAllByRole('heading', {level: 1}).innerHTML).toContain(tittle);
    expect( screen.getByRole('heading', {level: 1, name:tittle}));
  });

  test('dede de mostrar el subtitulo enviado por props',() =>{
    render(
      <FirstApp 
        title={tittle}
        subtitle={subtitle}
      />
    );
    expect(screen.getAllByText(subtitle).length).toBe(2);
  
  })
})
