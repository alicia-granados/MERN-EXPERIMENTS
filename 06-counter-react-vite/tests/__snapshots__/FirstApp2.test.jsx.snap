// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe de hacer match con el shapshot 1`] = `
<div>
  <h1>
    no hay tema
  </h1>
  <h1
    data-testid="test-title"
  >
    Hola soy goku
  </h1>
  <h2>
     
    2
     
  </h2>
  <p>
    Fernando
     
  </p>
  <code>
    {"message":"hola mundo","title":"Fernando"}
     
  </code>
  <p>
    Soy un subtitulo
  </p>
  <h1 />
  <h1 />
  <h1>
    Hola como estas
  </h1>
</div>
`;
